airflow/executors/base_executor.py

```diff
@@ -617,6 +622,7 @@ def debug_dump(self):
             len(self.event_buffer),
             "\n\t".join(map(repr, self.event_buffer.items())),
         )
+        self.log.debug("executor memory usgae (current: %d, peak: %d)", *tracemalloc.get_traced_memory())
```

Comment:
It's likely way more useful to display the top x allocations, instead of just current/peak, like [this](https://docs.python.org/3/library/tracemalloc.html#display-the-top-10).